### PR TITLE
refactor(rome_syntax): improve `inner_string_text`

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

This PR improves the concistency of the code-base. It introduces several changes:

- Remove `QuotedString` and `StaticStringValue` in favor of `TokenText` and `StaticValue`.
  This removes a bug in the implementation of `QuotedString` that was not reported yet.
- Remove unused functions related to these types
- Simplify `StaticValue` by replacing `TemplateChunk` with `String` and `EmptyString`
  - I removed dead (and misleading) code (comparison against `-0` and `+0` has no sense because we only handle positive numbers...)
  - I removed recognition of template string such as "`${1}`" because this leads to erroneous string values (for instance `${1n}` is converted to the string `1n` instead of the string `1`).

There is more refactoring to do (maybe removing `StaticValue` in favor of an union of AST nodes). However, there are already many changes. This seems enough for this PIR.

## Test Plan

Updated and new doc-tests.

### CODE REVIEW

File: crates/rome_js_syntax/src/static_value.rs [hunk 1]
```diff
@@ -98,8 +10,10 @@ pub enum StaticValue {
     Undefined(JsSyntaxToken),
     Number(JsSyntaxToken),
     BigInt(JsSyntaxToken),
-    String(QuotedString),
-    TemplateChunk(Option<JsSyntaxToken>),
+    /// The string can be empty.
```
Comment (ematipico) [inline crates/rome_js_syntax/src/static_value.rs:13]:
```suggestion
    // The string can be empty.
```

nit: It doesn't seem a doc comment relevant to the consumers

File: crates/rome_js_syntax/src/static_value.rs [hunk 2]
```diff
@@ -98,8 +10,10 @@ pub enum StaticValue {
...
+    String(JsSyntaxToken),
+    /// This is use to represent empty template strings.
```
Comment (ematipico) [inline crates/rome_js_syntax/src/static_value.rs:15]:
```suggestion
    /// This is used to represent empty template strings.
```

Is this valid also for cases like the following?

```js
`${variable}`
```

Technically it's "empty", in the sense that has empty strings inside.

File: crates/rome_js_syntax/src/static_value.rs [hunk 3]
```diff
@@ -109,22 +23,18 @@ impl StaticValue {
...
-            StaticValue::Null(_) => true,
-            StaticValue::Undefined(_) => true,
-            StaticValue::Number(token) => matches!(token.text_trimmed(), "0" | "-0" | "+0" | "NaN"),
-            StaticValue::BigInt(token) => matches!(token.text_trimmed(), "0n"),
-            StaticValue::String(token) => token.text().is_empty(),
-            StaticValue::TemplateChunk(token) => token
-                .as_ref()
-                .map_or(true, |it| it.text_trimmed().is_empty()),
+            StaticValue::Null(_) | StaticValue::Undefined(_) | StaticValue::EmptyString(_) => true,
+            StaticValue::Number(token) => token.text_trimmed() == "0",
```
Comment (ematipico) [inline crates/rome_js_syntax/src/static_value.rs:35]: What's the reason for the change to the variant `Number`?

**Review by ematipico — APPROVED** [review]
The best code is less code! 

Amazing refactors and cleaner code. You're amazing 🚀 

I left some questions and suggestions but overall we can merge it.

**Comment by Conaclos** [comment]
> The best code is less code!

As Antoine de Saint-Exupéry said:

> La perfection est atteinte, non pas lorsqu'il n'y a plus rien à ajouter, mais lorsqu'il n'y a plus rien à retirer.

Meaning: "Perfection is achieved, not when there is nothing more to add, but when there is nothing left to take away."

File: crates/rome_js_syntax/src/static_value.rs (hunk 2 above)
Comment (Conaclos) [reply crates/rome_js_syntax/src/static_value.rs:15]: No, this is just valid for the empty template string (without any element). I should rephrase to `the empty template string`

File: crates/rome_js_syntax/src/static_value.rs (hunk 3 above)
Comment (Conaclos) [reply crates/rome_js_syntax/src/static_value.rs:35]: `+0` and `-0` are not valid numbers.
I removed `NaN` because I think it is not the right way to handle `NaN`.
`NaN` is an identifier and could refer to a user-defined binding. Moreover, there is also `Number.NaN`.

We have the same issue with `undefined`. However, unit tests in aria related rules rely on this. It is the reason I kept `undefined`.
